src/Ormolu.hs

```diff
@@ -4,8 +4,8 @@
 -- | A formatter for Haskell source code.
 module Ormolu
   ( ormolu,
-    ormoluFile,
```

Comment:
I think we should go by the path of the minimal API change. For this kind of PR, I would expect the API to stay the same.  In general, it is a good idea to make your change least disruptive unless there is a good reason to refactor. IMO, having both `ormoluStdin` and `ormoluFile` is a reasonable choice and is marginally better than having an extra wrapper type just for this one function. As you pointed out elsewhere, the code in the docs is obsolete though and it is probably best to remove it altogether so that there is nothing in the docs that can go out of date easily. Also, this module is expected to be imported unqualified, this is why the main functions have `ormolu` in their names. `readAndFormat` is too general for a name. I understand this is a Haskell-specific naming habit and in languages like OCaml you would be indeed encouraged not to repeat module name in the names of functions.

Reply:
I agree with the minimal API change, but atm, when using ormolu on a file, this file needs to be read twice: the first time to store the original input (for diff later), and the second time in `ormoluFile` for formatting. In addition, I will be forced to change the API of `ormoluStdin`, because I need to store the original input (for diff in check mode), and AFAIK I can't read STDIN twice.
So in the first version I wrote (which I didn't commit), I just changed `ormoluStdin` to also returns the original input read from STDIN, but then there was almost a1:1 copy between code for file processing and STDIN processing, so I tried to improve it.
In fact, `ormoluStdin` and `ormoluFile` don't do much except hiding the original content read, which is useful in check mode, so I'm not sure what to do.

Reply:
Thanks, this is helpful. In that case let's keep the old `ormoluStdin` in the `Ormolu` module for people who potentially might want that function from calling from Haskell, but inline it in `Main.hs` this way we'll have the original input for comparison.